From 3dc4fd6dd41100f051a63642f449b16324389c96 Mon Sep 17 00:00:00 2001
From: Shengdun Wang <uwgghhbcad@gmail.com>
Date: Fri, 24 Apr 2026 23:23:54 -0700
Subject: [PATCH] [compiler-rt][sanitizer] Remove linux/scc.h

Linux removed the obsolete SCC headers.  Do not include their declarations
or expose their sizes from sanitizer_platform_limits_posix.cpp.

Upstream: 3dc4fd6dd41100f051a63642f449b16324389c96
---
 compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp | 21 ++++++++------------
 1 file changed, 8 insertions(+), 13 deletions(-)

diff --git a/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp b/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
index 1e7cfb3385dd..d6aa6c6d74d0 100644
--- a/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
+++ b/compiler-rt/lib/sanitizer_common/sanitizer_platform_limits_posix.cpp
@@ -152,16 +152,15 @@ typedef struct user_fpregs elf_fpregset_t;
 #if defined(__mips64)
 # include <sys/procfs.h>
 #endif
-#include <sys/user.h>
 #include <linux/if_eql.h>
 #include <linux/if_plip.h>
 #include <linux/lp.h>
 #include <linux/mroute.h>
 #include <linux/mroute6.h>
-#include <linux/scc.h>
 #include <linux/serial.h>
-#include <sys/msg.h>
 #include <sys/ipc.h>
+#include <sys/msg.h>
+#include <sys/user.h>
 #endif  // SANITIZER_ANDROID
 
 #include <link.h>
@@ -537,8 +536,6 @@ unsigned struct_ElfW_Phdr_sz = sizeof(Elf_Phdr);
 unsigned struct_kbsentry_sz = sizeof(struct kbsentry);
 unsigned struct_mtconfiginfo_sz = sizeof(struct mtconfiginfo);
 unsigned struct_nr_parms_struct_sz = sizeof(struct nr_parms_struct);
-unsigned struct_scc_modem_sz = sizeof(struct scc_modem);
-unsigned struct_scc_stat_sz = sizeof(struct scc_stat);
 unsigned struct_serial_multiport_struct_sz
     = sizeof(struct serial_multiport_struct);
 unsigned struct_serial_struct_sz = sizeof(struct serial_struct);
-- 
2.53.0
